sklearn/callback/tests/_utils.py

```diff
@@ -167,3 +165,86 @@ def _func(meta_estimator, inner_estimator, X, y, *, callback_ctx):
         estimator=meta_estimator,
         data={"X_train": X, "y_train": y},
     )
+
+
+class SimpleMetaEstimator(CallbackSupportMixin, BaseEstimator):
+    """A class that mimics the behavior of a meta-estimator that does not clone the
+    estimator and does not parallelize.
+    There is no iteration, the meta estimator simply calls the fit of the estimator once
+    in a subcontext.
```

Comment:
the fact that it doesn't parallelize has no impact I think. The important thing is the no cloning part so I'd give it a more explicit name.

```suggestion
class NoCloningMetaEstimator(CallbackSupportMixin, BaseEstimator):
    """A class that mimics the behavior of a meta-estimator that does not clone the
    estimator.
    There is no iteration, the meta estimator simply calls the fit of the estimator once
    in a subcontext.
```